### ISSUE TITLE
How to use buttons that jump to the document navigation bar on small form factor devices

### ISSUE THREAD

**Issue by ZhangChengLin** [twbs/bootstrap#38287]
https://github.com/twbs/bootstrap/blob/c2f67d72e8010307f657b2a55ae93fc5c9cab2a5/site/layouts/partials/skippy.html#L5

The current document navigation bar uses the offcanvas method, and when it is lower than (min-width: 768px), the document navigation bar is hidden.
The button to jump to the document navigation bar is not currently displayed on devices below (min-width: 768px), which does not seem perfect.
How do I make the button that jumps to the document navigation bar work on a small device?

**Comment by ZhangChengLin** [comment]
In addition, in the document page, do you want to add a button to jump to the table of contents, the specific location is the On this page of the page, and the anchor is #tocContents

**Comment by mootezaloui** [comment]
Hello,
To make the button that jumps to the document navigation bar work on small devices, you can add a conditional statement in the HTML code that displays the button when the device width is below the minimum threshold of 768px. Regarding your request to add a button that jumps to the table of contents, you can include an anchor link in the HTML code and add a button element that links to that anchor. I suggest placing the button in the "On this page" section of the page, and setting the link's href attribute to #tocContents. This will allow users to easily navigate to the table of contents on the same page

**Comment by ZhangChengLin** [comment]
> 

Yes, but you ignore that when the size is small, the bootstrap document navigation bar is hidden, click the jump to navigation bar button, will not trigger the display of the document navigation bar, so you should add the display of the trigger document navigation bar when you click the button, so that the focus can be positioned on the document navigation bar.
Currently, the bootstrap documentation does not do this. Instead, on small form factor devices (min-width: 768px), the jump to document navigation bar button does not appear.

**Comment by mootezaloui** [comment]
It looks that the problem is not just with showing the button, but also with making sure that when the button is clicked on tiny devices, the document navigation bar is visible. If so, you may change the JavaScript code to include an event listener on the button element, which when clicked toggles the visibility of the document navigation bar.
For example, you could add something like this to the code: 
`document.querySelector('#jump-to-nav-button').addEventListener('click', function() {`
`  var navBar = document.querySelector('#document-nav-bar');`
` if (navBar.classList.contains('offcanvas')) {`
` navBar.classList.remove('offcanvas');`
` navBar.classList.add('show');`
`  } else {`
` navBar.classList.add('offcanvas');`
` navBar.classList.remove('show');`
`}`
`});`
This code listens for a click on the button with the ID "jump-to-nav-button", then toggles the "offcanvas" and "show" classes on the document navigation bar element with the ID "document-nav-bar". This should ensure that the navigation bar is displayed when the button is clicked on small devices. 
However You are true that the document navigation bar does not appear when the hop to navigation button is clicked on tiny form factor devices according to the current Bootstrap instructions. When the button is clicked, JavaScript may be used to alter the navigation bar's appearance in order to provide this feature. For this think this may work better 
`const jumpToNavButton = document.querySelector('#jump-to-docs-nav');`
`const navBar = document.querySelector('#bd-docs-nav');`
`jumpToNavButton.addEventListener('click', () => {`
  `if (window.innerWidth < 768) {`
   ` navBar.style.display = 'block';`
  `}`
`});`
The jumpToNavButton variable is set to the element representing a jump to navigation button, and the navBar variable is assigned to the element representing a document navigation bar. When a button is pressed, the event listener determines whether the device width is less than 768 pixels and, if it is, changes the navigation bar's display to "block" mode.

I hope this 2 solutions works for you.

**Comment by ZhangChengLin** [comment]
> 

`    <a class="d-inline-flex p-2 m-1" href="#bd-docs-nav" data-bs-toggle="offcanvas" data-bs-target="#bdSidebar">Skip to docs navigation</a>
`
